### PR TITLE
Completed task and passed test. Hibernate relations.

### PR DESCRIPTION
**Pull request by aksoli666**
-added required hibernate and sql dependencies;
-completed implementation of all classes in model package; 
-completed implementation of all classes in dao.impl package; 
-created custom unchecked DataProcessingException; 
-completed implementation of all classes in service.impl package; 
-created directory resources with hibernate.cfg.xml file; 
-corrected code in Main.main().

### CODE REVIEW

File: src/main/java/mate/academy/hibernate/relations/model/Actor.java [hunk 1]
```diff
@@ -1,8 +1,18 @@
...
+import javax.persistence.Id;
+import javax.persistence.ManyToOne;
+
+@Entity
 public class Actor implements Cloneable {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
     private Long id;
     private String name;
+    @ManyToOne()
```
Comment (Rommelua) [inline src/main/java/mate/academy/hibernate/relations/model/Actor.java:15]:
```suggestion
    @ManyToOne
```

File: src/main/java/mate/academy/hibernate/relations/model/Movie.java [hunk 2]
```diff
@@ -2,10 +2,20 @@
...
+import javax.persistence.Id;
+import javax.persistence.OneToMany;
 
+@Entity
 public class Movie implements Cloneable {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
     private Long id;
     private String title;
+    @OneToMany(fetch = FetchType.EAGER)
```
Comment (Rommelua) [inline src/main/java/mate/academy/hibernate/relations/model/Movie.java:18]: think about relation type

File: src/main/java/mate/academy/hibernate/relations/service/impl/ActorServiceImpl.java [hunk 3]
```diff
@@ -1,16 +1,26 @@
...
 import mate.academy.hibernate.relations.service.ActorService;
+import org.hibernate.SessionFactory;
 
 public class ActorServiceImpl implements ActorService {
+    private SessionFactory sessionFactory;
+    private final ActorDao actorDao = new ActorDaoImpl(sessionFactory);
+
+    public ActorServiceImpl(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
```
Comment (Rommelua) [inline src/main/java/mate/academy/hibernate/relations/service/impl/ActorServiceImpl.java:15]: For all services
```suggestion
    private final ActorDao actorDao;

    public ActorServiceImpl(ActorDao actorDao) {
        this.actorDao = actorDao ;
    }
```

**Review by Rommelua — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/hibernate/relations/service/impl/ActorServiceImpl.java [hunk 4]
```diff
@@ -1,16 +1,23 @@
...
     @Override
     public Actor add(Actor actor) {
-        return null;
+        return actorDao.add(actor);
     }
 
     @Override
     public Actor get(Long id) {
-        return null;
+        return actorDao.get(id).get();
```
Comment (MentorMA) [inline src/main/java/mate/academy/hibernate/relations/service/impl/ActorServiceImpl.java:21]: just using method get() from Optional is bad practice, better use orElseThrow

File: src/main/java/mate/academy/hibernate/relations/service/impl/CountryServiceImpl.java [hunk 5]
```diff
@@ -1,16 +1,23 @@
...
     @Override
     public Country add(Country country) {
-        return null;
+        return countryDao.add(country);
     }
 
     @Override
     public Country get(Long id) {
-        return null;
+        return countryDao.get(id).get();
```
Comment (MentorMA) [inline src/main/java/mate/academy/hibernate/relations/service/impl/CountryServiceImpl.java:21]: Same situation

**Review by MentorMA — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/hibernate/relations/service/impl/MovieServiceImpl.java [hunk 6]
```diff
@@ -1,16 +1,23 @@
...
     @Override
     public Movie add(Movie movie) {
-        return null;
+        return movieDao.add(movie);
     }
 
     @Override
     public Movie get(Long id) {
-        return null;
+        return movieDao.get(id).get();
```
Comment (kerrrusha) [inline src/main/java/mate/academy/hibernate/relations/service/impl/MovieServiceImpl.java:21]: `.orElseThrow` should be used instead

File: src/main/java/mate/academy/hibernate/relations/service/impl/CountryServiceImpl.java [hunk 7]
```diff
@@ -1,16 +1,25 @@
...
-        return null;
+        return countryDao.add(country);
     }
 
     @Override
     public Country get(Long id) {
-        return null;
+        return countryDao.get(id).orElseThrow(()
+                -> new RuntimeException("There is no way to get Country by id: "
+                + id + ". That you get null"));
```
Comment (kerrrusha) [inline src/main/java/mate/academy/hibernate/relations/service/impl/CountryServiceImpl.java:23]:
```suggestion
                -> new RuntimeException("Can't find country by id: " + id ));
```

File: src/main/java/mate/academy/hibernate/relations/service/impl/ActorServiceImpl.java [hunk 8]
```diff
@@ -1,16 +1,25 @@
...
-        return null;
+        return actorDao.add(actor);
     }
 
     @Override
     public Actor get(Long id) {
-        return null;
+        return actorDao.get(id).orElseThrow(()
+                -> new RuntimeException("There is no way to get Actor by id: "
+                + id + ". That you get null"));
```
Comment (kerrrusha) [inline src/main/java/mate/academy/hibernate/relations/service/impl/ActorServiceImpl.java:23]:
```suggestion
                -> new RuntimeException("Can't find actor by id: " + id));
```

**Review by kerrrusha — CHANGES_REQUESTED** [review]

**Review by MentorMA — APPROVED** [review]

**Pull request closed without merging**